lbz/authz.py

```diff
@@ -25,106 +21,74 @@
 LIMITED_ALLOW = -1
 
 
-class Authorizer(metaclass=Singleton):
-    allow = {}
-    deny = {}
-    action = None
-    outcome = None
-    allowed_resource = None
-    denied_resource = None
-    expiration = None
-    iss = None
-
-    def __init__(self, resource=None):
-        self.resource = resource
-        self._permissions = NestedDict()
-
-    def __getitem__(self, y):
-        return self._permissions[y]
-
-    def __str__(self):
-        return json.dumps({self.resource: self._permissions})
+class Authorizer:
+    def __init__(self, auth_jwt: str, resource_name: str, permission_name: str):
+        self.outcome = DENY
```

Comment:
its better to have them as class variables as they are not being changed during init.

Reply:
I'd say logically they should be declared in init - they are attributes of an instance, not of the whole class. Declaring instance attributes as class attributes can lead to problems, I don't remember specific cases now, but I know I had to deal it at some point in the past.

Reply:
Ok, here is an example:
```
class Abc:
    aaa = []

    def inc(self):
        self.aaa.append(1)

q = Abc()
w = Abc()
print(w.aaa)
q.inc()
print(w.aaa)
```
the second print will print `[1]` which is not what we expect since we never run `inc()` on `w`

Reply:
ACK